### PR TITLE
Update Phoenix.Logger docs

### PR DESCRIPTION
**Pull request by qcam**

### CODE REVIEW

**Comment by chrismccord** [comment]
This doesn't appear to be something we emit from telemetry. What are you seeing `result` come from? Thanks!

File: lib/phoenix/logger.ex [hunk 1]
```diff
@@ -69,12 +69,12 @@ defmodule Phoenix.Logger do
 
     * `[:phoenix, :channel_joined]` - dispatched at the end of a channel join
       * Measurement: `%{duration: native_time}`
-      * Metadata: `%{params: term, socket: Phoenix.Socket.t}`
+      * Metadata: `%{result: :ok | :error, params: term, socket: Phoenix.Socket.t}`
```
Comment (qcam) [inline lib/phoenix/logger.ex:72]: I think we emit `:result` at https://github.com/phoenixframework/phoenix/blob/ac35fba991d550494c9b788a9d57bbf5679a3dcf/lib/phoenix/channel/server.ex#L301.

Comment (qcam) [reply lib/phoenix/logger.ex:72]: I was wrong about the other 🙈. It does not seem to emit `:result`.

Comment (chrismccord) [reply lib/phoenix/logger.ex:72]: and I only checked the other 🙈 :D

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**